### PR TITLE
[Blog Post] Generate Dynamic JSON Pages with Next.js

### PR DESCRIPTION
**Pull request by seancdavis**

### CODE REVIEW

File: pages/blog/dynamic-json-pages-with-nextjs.mdx [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+The first method is powerful and all, but it's also forcing you into a solution in which you have to run that method (i.e. do some work, like hit another API) every time you want the data in this file.
+
+Recall in the intro that I mentioned a nuance of not needing the file to change in between builds. Thus, the JSON file itself should be _generated dynamically_, but could be _delivered statically_. (This has all the makings of a [static API](https://www.seancdavis.com/blog/lets-talk-about-static-apis/).)
+
+While we know we can't make pages as static JSON files, we could _generate_ a static JSON file prior to building the site and serve it [as a static asset](https://nextjs.org/docs/basic-features/static-file-serving).
+
+### Generate Script
+
+To do that, let's put together a little script at `scripts/getJoke.js` which retrieves the dad joke and then writes it to a file at `public/joke.json`.
```
Comment (evantahler) [inline pages/blog/dynamic-json-pages-with-nextjs.mdx:121]: Maybe a note about why it has to be stored in `/public` would be helpful

**Review by evantahler — APPROVED** [review]
Nice!

File: pages/blog/dynamic-json-pages-with-nextjs.mdx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+And the server runs at localhost:3000 in your browser.
+
+If you had an existing Next project, you may have a different command to start the dev server and a different port on which the front end runs.
+
+## Method #1: API Routes
+
+Now that you're up and running, let's look at our first option for generating JSON pages: API routes.
+
+Okay, I lied. A little. Pages in next don't _have to be_ React components. Next also supports what they call [API routes](https://nextjs.org/docs/api-routes/introduction). These are methods that run on the server side and return data base to the user. That feels like a really good use case for our scenario.
```
Comment (teallarson) [inline pages/blog/dynamic-json-pages-with-nextjs.mdx]: I'm wondering if this would read smoother as "...return data back to the user"?

**Review by teallarson — APPROVED** [review]
This is great!  I left one comment about what was either a phrase I'm not familiar with or was a typo.

**Review by pedroslopez — APPROVED** [review]
I like this!

**Pull request merged**